server/database/repositories/TournamentRepository.ts

```diff
@@ -1,37 +1,114 @@
+      score: {
+        a: matchDoc.sideA.score.a,
+        b: matchDoc.sideA.score.b,
+      },
+    },
+    isFinished: matchDoc.isFinished,
+  };
+  return match;
+}
+interface TournamentDb extends Omit<TournamentWihtoutMS, 'id' | 'teams' | 'matches'> {
+  _id?: string,
+  teams: TeamDb[],
+  matches: MatchDb[]
+}
+function toTournamentDb(t: TournamentWihtoutMS) : TournamentDb {
+  const tDb : TournamentDb = {
+    _id: t.id,
+    name: t.name,
+    ownerId: t.ownerId,
+    teams: t.teams.map((team) => toTeamDb(team)),
+    matches: t.matches.map((match) => toMatchDb(match)),
+  };
+  return tDb;
+}
+function toTournament(tourDoc: TournamentDocument) : TournamentWihtoutMS {
+  const tournament : TournamentWihtoutMS = {
+    id: tourDoc._id.toString(),
+    ownerId: tourDoc.ownerId,
+    name: tourDoc.name,
+    teams: tourDoc.teams.map((teamDoc) => toTeam(teamDoc)),
+    matches: tourDoc.matches.map((matchDoc) => toMatch(matchDoc)),
+  };
+  return tournament;
+}
 
 export default class TournamentRepository {
   public static async create(tournament: TournamentWihtoutMS) {
-    // Check if teams have ids, if teams in matches have ids
-    // and if teams in matches are on teams list
+    if (tournament.id) { throw new AppError('New tournament should not have id', 400); }
     tournament.teams.forEach((team) => {
-      if (!team._id) { throw new AppError('All teams should have ids.', 400); }
+      if (!team.id) { throw new AppError('All teams should have ids.', 400); }
     });
     tournament.matches.forEach((match) => {
-      if (!match.sideA.team._id || !match.sideB.team._id) { throw new AppError('All teams in matches entries should have ids', 400); }
+      if (!match.sideA.team.id || !match.sideB.team.id) { throw new AppError('All teams in matches entries should have ids', 400); }
     });
-
-    return await TournamentModel.create(tournament);
+    const tDoc = await TournamentModel.create(toTournamentDb(tournament));
+    const tour = await TournamentRepository.getById(tDoc._id);
+    if (!tour) { throw new AppError('Error while creating tournament', 400); }
+    return tour;
   }
 
-  public static async updateMatch(matchId: string, match: Match) {
-    if (!matchId) throw new AppError('Provided match does not contain id', 400);
-
-    const tournament = await TournamentModel.findOneAndUpdate(
-      { 'matches._id': matchId },
+  public static async updateMatch(match: MatchWithoutMS) {
+    if (!match.id) throw new AppError('Provided match does not contain id', 400);
+    const matchDb = toMatchDb(match);
+    const tDoc: TournamentDocument | null = await TournamentModel.findOneAndUpdate(
+      { 'matches._id': matchDb._id },
       {
         $set: {
-          'matches.$': match,
+          'matches.$': matchDb,
         },
       },
       { new: true },
-    ).lean();
-
-    return tournament;
+    ).exec();
+    if (!tDoc) throw new AppError('Attempting to update a non-existent match', 400);
```

Comment:
```suggestion
    if (!tDoc) throw new AppError('Match does not exits', 404);
```
But is necessary to checking this? I wonder if we should check this before calling this method in business logic

Reply:
If we decide to update it, we have to be sure that specific match exists, we can return null in method like : getMatchById, and then we can check if match is null in b.log.

Reply:
What's the point of trying to update match that we even don't know if it exists